src/coreclr/src/jit/lowerarmarch.cpp

```diff
@@ -535,8 +535,255 @@ void Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
         node->gtType = TYP_SIMD16;
     }
 
+    NamedIntrinsic intrinsicId = node->gtHWIntrinsicId;
+
+    switch (intrinsicId)
+    {
+        case NI_Vector64_Create:
+        case NI_Vector128_Create:
+        {
+            // We don't directly support the Vector64.Create or Vector128.Create methods in codegen
+            // and instead lower them to other intrinsic nodes in LowerHWIntrinsicCreate so we expect
+            // that the node is modified to either not be a HWIntrinsic node or that it is no longer
+            // the same intrinsic as when it came in.
+
+            LowerHWIntrinsicCreate(node);
+            assert(!node->OperIsHWIntrinsic() || (node->gtHWIntrinsicId != intrinsicId));
+            LowerNode(node);
+            return;
+        }
+
+        default:
+            break;
+    }
+
     ContainCheckHWIntrinsic(node);
 }
+
+//----------------------------------------------------------------------------------------------
+// Lowering::LowerHWIntrinsicCreate: Lowers a Vector64 or Vector128 Create call
+//
+//  Arguments:
+//     node - The hardware intrinsic node.
+//
+void Lowering::LowerHWIntrinsicCreate(GenTreeHWIntrinsic* node)
+{
+    NamedIntrinsic intrinsicId = node->gtHWIntrinsicId;
+    var_types      simdType    = node->gtType;
+    var_types      baseType    = node->gtSIMDBaseType;
+    unsigned       simdSize    = node->gtSIMDSize;
+    VectorConstant vecCns      = {};
+
+    if ((simdSize == 8) && (simdType == TYP_DOUBLE))
+    {
+        simdType = TYP_SIMD8;
```

Comment:
This is because `impFixupStructReturnType` changes the node from `TYP_SIMD8` to `TYP_DOUBLE`. I'd imagine it would be desirable to preserve it as `TYP_SIMD8` and adjust the various code paths to recognize `TYP_SIMD8`. 

Is that covered by an existing issue?

Reply:
This is related to the work that @sandreenko is doing to avoid retyping return types.